src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -38,15 +44,12 @@ public override bool IsWhiteSpace
         {
             get
             {
-                foreach (var value in _buffer)
+                if (_buffer == null)
                 {
-                    if (!string.IsNullOrWhiteSpace(value))
-                    {
-                        return false;
-                    }
+                    return true;
                 }
 
-                return true;
+                return CheckIfBufferedHtmlContentIsWhiteSpace(Buffer);
```

Comment:
No need for the indirection through the property here or in a few other places in this class. Suggest sticking with the field (`_buffer`) after checking it for null.